src/sheldon_suggestions_server.erl

```diff
@@ -0,0 +1,70 @@
+-module(sheldon_suggestions_server).
+
+-behaviour(gen_server).
+
+%%% API
+-export([ suggestions/2
+        , add_suggestions/2
+        ]).
+
+%%% gen_server callbacks
+-export([
+  init/1,
+  handle_call/3,
+  handle_cast/2,
+  handle_info/2,
+  terminate/2,
+  code_change/3
+]).
+
+%%%=============================================================================
+%%% API
+%%%=============================================================================
+
+-spec suggestions([sheldon_result:misspelled_word()], sheldon_dictionary:language()) ->
+  [sheldon_result:misspelled_word()].
+suggestions(MisspelledWords, Lang) ->
+  try rpc:pmap({?MODULE, add_suggestions}, [Lang], MisspelledWords) of
+    Result -> Result
+  catch
+    exit:badrpc -> MisspelledWords
+  end.
+
+-spec add_suggestions(sheldon_result:misspelled_word(), sheldon_dictionary:language()) ->
+  sheldon_result:misspelled_word().
+add_suggestions(#{word := Word} = MisspelledWord, Lang) ->
+  Candidates = wpool:call(suggestions_pool, {suggest, Word, Lang}, available_worker, infinity),
```

Comment:
Don't use `infinity`… leave it 5 seconds. Worst case scenario, you get a `timeout` and produce no candidates for the word. That wouldn't be too bad.